File: app/src/main/java/ru/nsu/fit/modao/adapter/ParticipantsAdapter.java
package ru.nsu.fit.modao.adapter;

import android.annotation.SuppressLint;
import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.LinkedList;
import java.util.List;

import ru.nsu.fit.modao.R;
import ru.nsu.fit.modao.fragments.EnterCoefficientsFragment;
import ru.nsu.fit.modao.model.CreateExpenseParticipant;
import ru.nsu.fit.modao.model.User;

public class ParticipantsAdapter extends RecyclerView.Adapter<ParticipantsAdapter.ParticipantsViewHolder> {
    Context context;
    List<CreateExpenseParticipant> participants;
    List<CreateExpenseParticipant> selected;

    public ParticipantsAdapter(Context context, List<CreateExpenseParticipant> participants) {
        this.context = context;
        this.participants = participants;
        selected = new LinkedList<>();
    }

    @NonNull
    @Override
    public ParticipantsViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View participantItem = LayoutInflater.from(context).inflate(R.layout.participant_item, parent, false);
        return new ParticipantsAdapter.ParticipantsViewHolder(participantItem);
    }

    @Override
    public void onBindViewHolder(@NonNull ParticipantsViewHolder holder, @SuppressLint("RecyclerView") int position) {
        holder.nameParticipant.setText(participants.get(position).getUsername());
        holder.checkName.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                if (holder.checkName.isChecked()){
                    selected.add(participants.get(position));
                }
                else{
                    selected.remove(participants.get(position));
                }
            }
        });
    }

    @Override
    public int getItemCount() {
        return participants.size();
    }

    public static final class ParticipantsViewHolder extends RecyclerView.ViewHolder{

        TextView nameParticipant;
        CheckBox checkName;

        public ParticipantsViewHolder(@NonNull View itemView) {
            super(itemView);

            nameParticipant = itemView.findViewById(R.id.nameParticipant);
            checkName = itemView.findViewById(R.id.checkParticipant);
        }
    }
    public List<CreateExpenseParticipant> getSelected(){
        return selected;
    }
}

File: app/src/main/java/ru/nsu/fit/modao/adapter/ParticipantsAdapter.kt
package ru.nsu.fit.modao.adapter

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CompoundButton
import android.widget.RadioButton
import androidx.recyclerview.widget.RecyclerView
import ru.nsu.fit.modao.R
import ru.nsu.fit.modao.databinding.ParticipantItemBinding
import ru.nsu.fit.modao.databinding.ParticipantSpentItemBinding
import ru.nsu.fit.modao.models.ParticipantEvent

class ParticipantsAdapter: RecyclerView.Adapter<ParticipantsAdapter.ParticipantsHolder>() {
    private var listener: CustomListener? = null
    private lateinit var list: Array<ParticipantEvent>
    private var initListener: InitListener? = null

    fun attachListener(listener: CustomListener){
        this.listener = listener
    }
    fun attachInitListener(initListener: InitListener){
        this.initListener = initListener
    }
    fun setList(list: Array<ParticipantEvent>){
        this.list = list
        notifyDataSetChanged()
    }
    class ParticipantsHolder(item: View): RecyclerView.ViewHolder(item){
        val binding = ParticipantSpentItemBinding.bind(item)
        fun bind(user: ParticipantEvent, listener: CustomListener, initListener: InitListener){
            binding.nameParticipant.text = user.username
            binding.selectParticipant.setOnClickListener(){
                listener.onClickItem(it as RadioButton, user)
            }
            binding.selectParticipant.isChecked = user.isSponsor
            if (user.isSponsor){
                initListener.initItem(binding.selectParticipant, user)
            }
        }
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ParticipantsHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.participant_spent_item, parent, false)
        return ParticipantsHolder(view)
    }

    override fun onBindViewHolder(holder: ParticipantsHolder, position: Int) {
        holder.bind(list[position], listener!!, initListener!!)
    }

    override fun getItemCount(): Int {
        return list.size
    }
    interface CustomListener{
        fun onClickItem(button: RadioButton, user: ParticipantEvent)
    }
    interface InitListener{
        fun initItem(button: RadioButton, user: ParticipantEvent)
    }
}